### ISSUE TITLE
defroutes* and response messages fail together

### ISSUE THREAD

**Issue by leon-barrett** [metosin/compojure-api#96]
When I use defroutes* and :responses with messages, as described in the README, the resulting swagger output does not have descriptions. It looks like something is going wrong in the chain of macros (maybe metadata is getting stripped somehow?).

The issue is demonstrated with the following test. I'll follow up with a pull request for this test.

```
(deftest test-response-descriptions
  (defroutes* response-descriptions-routes
    (GET* "/x" []
          :responses {500 ^{:message "Server error"} s/Str}
          :return s/Str
          "x"))
  (defapi response-descriptions-api
    (swagger-docs)
    (context* "/api" []
              :tags ["api"]
              response-descriptions-routes))
  (fact (let [[status spec]
              (get* response-descriptions-api "/swagger.json" {})]
          status => 200
          (-> spec :paths vals first :get :responses :500 :description)
          => "Server error")))
```

**Comment by ikitommi** [comment]
thanks for the test, will try to fix that soon.

**Comment by ikitommi** [comment]
Meta-data attached by the reader macro doesn't work well with compojure-api's source linking. Created and tested a (ugly) fix this, but won't propably merge it back to master.

the ugly fix is to use normal `describe`:

```clojure
(defroutes* response-descriptions-routes
  (GET* "/x" []
    :responses {500 (describe s/Str "Server error")}
    identity))
```

better solution would be to start using the Ring-Swaggers 2.0 format:

```clojure
{:responses {200 {:schema Parrot
                  :description ""
                  :headers ...}}}
```

would be a breaking change, but use of old format could be detected in 99% cases with compile-time descriptive exception.

**Comment by ikitommi** [comment]
Added support for setting the meta-data via normal function `(with-meta)`. See https://github.com/metosin/compojure-api#response-models for an example. There is also a test to verify this:

https://github.com/metosin/compojure-api/blob/1daa0f1a5436f820bccdb985ef797f877f0919ca/test/compojure/api/core_integration_test.clj#L877-L889

changes are found in:

```clojure
[metosin/compojure-api "0.20.4-SNAPSHOT"]
```

**Comment by leon-barrett** [comment]
Thanks!

**Comment by ikitommi** [comment]
published `0.20.4` with this and snapshotted `0.21.0` with the final model. Should be a smooth transition to it - old format will give [helpful compile-time errors](https://github.com/metosin/compojure-api/blob/e05071c57c9fcf78423af35f7335ccce30cbf924/src/compojure/api/meta.clj#L87-L95):

https://github.com/metosin/compojure-api#response-models-pre-0210 ->
https://github.com/metosin/compojure-api#response-models-post-0210
